docs/proposals/block-storage-time-series-deletion.md

```diff
@@ -0,0 +1,247 @@
+
+
+This can be very important for users to have as confidential or accidental data might have been incorrectly pushed and needs to be removed. As well as potentially removing high cardinality data that is causing inefficient queries. 
+
+## Related works
+
+As previously mentioned, the deletion feature is already implemented with chunk storage. The main functionality is implemented through the purger service. It accepts requests for deletion and processes them.  At first, when a deletion request is made, a tombstone is created. This is used to filter out the data for queries. After some time, a deletion plan is executed where the data is permanently removed from chunk storage. 
+
+Can find more info here:
+
+- [Cortex documentation for chunk store deletion](https://cortexmetrics.io/docs/guides/deleting-series/)
+- [Chunk deletion proposal](https://docs.google.com/document/d/1PeKwP3aGo3xVrR-2qJdoFdzTJxT8FcAbLm2ew_6UQyQ/edit)
+
+
+
+## Background on current storage
+
+With a block-storage configuration, Cortex stores data that could be potentially deleted by a user in:
+
+- Object store (GCS, S3, etc..) for long term storage of blocks
+- Ingesters for more recent data that should be eventually transferred to the object store
+- Cache 
+    - Index cache
+    - Metadata cache
+    - Chunks cache (stores the potentially to be deleted data) 
+    - Query results cache  (stores the potentially to be deleted data) 
+- Compactor during the compaction process
+- Store-gateway
+
+
+## Proposal
+
+The deletion will not happen right away. Initially, the data will be filtered out from queries using tombstones and will be deleted afterward. This will allow the user some time to cancel the delete request. 
+
+### API Endpoints
+
+The existing purger service will be used to process the incoming requests for deletion. The API will follow the same structure as the chunk storage endpoints for deletion, which is also based on the Prometheus deletion API. 
+
+This will enable the following endpoints for Cortex when using block storage:
+
+`POST /api/v1/admin/tsdb/delete_series` - Accepts [Prometheus style delete request](https://prometheus.io/docs/prometheus/latest/querying/api/#delete-series)  for deleting series.
+
+Parameters:
+
+- `start=<rfc3339 | unix_timestamp>`
+    - Optional. If not provided, will be set to minimum possible time. 
+- `end=<rfc3339 | unix_timestamp> `
+    - Optional. If not provided, will be set to maximum possible time (time when request was made). End time cannot be greater than the current UTC time. 
+- `match[]=<series_selector>`
+    - Cannot be empty, must contain at least one label matcher argument. 
+
+
+`POST /api/v1/admin/tsdb/cancel_delete_request`  - To cancel a request if it has not been processed yet for permanent deletion. This can only be done before the `-purger.delete-request-cancel-period` has passed. 
```

Comment:
How exactly does the cancellation work? Will it just delete the tombstone, or update it? Or just change its state?

Reply:
Overall, I propose to delete the tombstone and invalidate the cache. I added a bit more detail regarding the cache invalidation in the proposal using the solution you proposed in an earlier comment.